packages/react-client/src/hooks/useTrackManager.ts

```diff
@@ -69,7 +75,7 @@ export const useTrackManager = ({ mediaManager, tsClient, getCurrentPeerStatus }
   }
 
   const startStreaming = useCallback(
-    async (simulcastConfig?: SimulcastConfig, maxBandwidth?: TrackBandwidthLimit) => {
+    async (props: StartStreamingProps = { simulcastConfig: DEFAULT_SIMULCAST_CONFIG }) => {
```

Comment:
Since it's a client facing API I would suggest simplifying it:
1. It's possible to invoke it like that, which doesn't make any sense.

```ts
 const defaultConfig = new Map<Encoding, number>([
        ['l', 10],
        ['m', 99999],
        ['h', 0],
      ]);

      const props2: StartStreamingProps = {
        simulcastConfig: { enabled: false, activeEncodings: [], disabledEncodings: [] },
        maxBandwidth: defaultConfig
      };

```

2. Simulcast config is unnecessarily complicated

```ts
export interface SimulcastConfig {
  enabled: boolean;
  activeEncodings: Encoding[];
  disabledEncodings: Encoding[];
}
``` 
  a) If someone passes a simulcast config it means that they want to use simulcast, so `enabled` is redundant
  b) We could merge together active and disabled encodings into one record:

```ts
const encoding: Record<Encoding, boolean> = {
  h: true,
  m: false,
  l: false,
};

 ```

c) `TrackBandwidthLimit` which could be `SimulcastBandwidthLimit` is too complicated. I think we could replace it with a simple record

```ts
export type SimulcastBandwidthLimit = Map<Encoding, BandwidthLimit>;
export type TrackBandwidthLimit = BandwidthLimit | SimulcastBandwidthLimit;

// from this:
export const bitrate: SimulcastBandwidthLimit = new Map<Encoding, number>([
  ['l', 2_500_000],
  ['m', 500_000],
  ['h', 150_000],
]);
 // to this
 export const bitrate: Record<Encoding, number> = {
  h: 2_500_000,
  m: 500_000,
  l: 150_000,
};
```


I would suggest to use a simpler structure. Two ideas:

```ts
type SimulcastEncodings = Partial<Record<Encoding, boolean>>
type SimulcastMaxBandwidth = Partial<Record<Encoding, number>>

type SimulcastConfig = { maxBandwidth?: SimulcastMaxBandwidth, encodings?: SimulcastEncodings }
type NonSimulcastConfig = { maxBandwidth: number, encodings?: never; }

type Props = SimulcastConfig | NonSimulcastConfig

const simulcast1: Props = { maxBandwidth: { l: 0, h: 0 } }
const simulcast2: Props = { encodings: { l: true, h: false } }
const simulcast3: Props = { maxBandwidth: { l: 0, m: 0, h: 0 }, encodings: { l: true, h: false } }

const nonSimulcast: Props = { maxBandwidth: 20 }

const error: Props = { maxBandwidth: 20, encodings: { l: true, h: false } } // error
```

It's not perfect because something like `simulcast3` that is still possible

Another second idea:

```ts
type SimulcastLayer = "unlimited" | number | "disabled"
type SimulcastConfig = Partial<Record<Encoding, SimulcastLayer>>
type NonSimulcastConfig = { maxBandwidth: number, h?: never, m?: never, l?: never }

type Props = SimulcastConfig | NonSimulcastConfig

const simulcast1: Props = { h: "disabled", m: "unlimited", l: 25_000 }
const nonSimulcast: Props = { maxBandwidth: 25_000 }
const error: Props = { maxBandwidth: 25_000, h: 25_000 }
```